server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -80,8 +80,22 @@ class InteractiveSession(
 
     def mergeConfList(list: Seq[String], key: String): Unit = {
       if (list.nonEmpty) {
-        val newList = (list ++ conf.get(key)).mkString(",")
-        builder.setConf(key, newList)
+        val oldList = builder.getConf(key)
```

Comment:
That sounds like a problem that can be solved without changing the `LivyClientBuilder`. I just don't want to expose public methods (and `LivyClientBuilder` is a public class in Livy) that are not really necessary.